ghproxy/ghcache/ghcache.go

```diff
@@ -234,6 +239,13 @@ func (tr *tokensRegistry) calculateRequestWaitDuration(lastRequest tokenInfo, to
 		}
 		future := lastRequest.timestamp.Add(difference)
 		duration = future.Sub(toQueue)
+
+		// There is so long queue of messages scheduled that we have to assign max waiting time to enable the possibility
+		// of quick recovery when the queue starts to be consumed again (this use-case is a result of very rare GH API degradation incidents)
```

Comment:
```suggestion
		// Do not exceed max wait time to avoid creating a huge request backlog if the GitHub api has performance issues
```

Is the above correct? This happens because requests takes very long and the request execution can not keep up with the rate of incoming request?

Reply:
...and they are scheduled with time spacing and they are not handled, queue grows. Yes.